helix-lsp/src/transport.rs

```diff
@@ -125,19 +126,21 @@ impl Transport {
             }
         }
 
-        let content_length = content_length.context("missing content length")?;
+        let len = len.context("missing content length")?;
+        message.resize(len, 0);
 
-        //TODO: reuse vector
```

Comment:
Resolving this TODO is good either way. I've ported out this change to master and applied it to the DAP transport as well in 8d2870b94af238a30dc12e14d8fd19c06a202f9d